src/lib/y2storage/storage_features_list.rb

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) [2016-2017,2019-2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "forwardable"
+require "y2storage/storage_feature"
+
+module Y2Storage
+  # List of storage features
+  #
+  # In libstorage-ng a set of features is representend by an integer
+  # bit-field that must be processed to get the list of features, each one
+  # represented by a bit-mask referenced by a constant.
+  #
+  # This class turns one of those bit-fields into an object-oriented collection
+  # of {StorageFeature} objects.
+  class StorageFeaturesList
+    include Yast::Logger
+    include Enumerable
+    extend Forwardable
+
+    def_delegators :@features, :each, :empty?
+
+    # Constructor
+    #
+    # @param bits [Integer, nil] bit-field representing all the features that
+    #   must be part of the list. If nil, the list will contain all the known
+    #   storage features.
+    def initialize(bits = nil)
+      @features = StorageFeature.all.dup
```

Comment:
np: I usually prefer passing the list of features as an argument as it makes it easier to test even without mocking. However, in this case, the constructor would look even more weird :).

Reply:
Fixed in a later commit already added to this PR.